npeps/npep-28-egress-traffic-control.md

```diff
@@ -0,0 +1,74 @@
+# NPEP-28: Add northbound traffic support in (B)ANP API
+
+* Issue: [#28](https://github.com/kubernetes-sigs/network-policy-api/issues/28)
+* Status: Provisional
+
+## TLDR
+
+This NPEP proposes adding support for cluster egress (northbound) traffic control in the `AdminNetworkPolicy` and `BaselineAdminNetworkPolicy` API objects.
+
+## Goals
+
+* Implement egress traffic control towards external destinations (outside the cluster)
+* Implement egress traffic control towards cluster nodes and/or host-networked pods
```

Comment:
I think we can discuss it here! 

> So we probably need to decide if we just say "hostNetwork pod and host traffic are indistinguishable" or if we want to have more options.

I think it would make sense to start with something like a "node selector"... Then it's explicitly easy for plugins to say we don't support it or we do.  Ultimately for the first round I think a node/host networked pod _should_ be treated the same since essentially they will always have the same IP ( I think )

Reply:
checking with calico/cilium ones https://docs.cilium.io/en/latest/security/policy/language/#hostpolicies and https://docs.tigera.io/calico/latest/reference/resources/globalnetworkpolicy#selectors it seems we don't distinguish between the two.. so let's go with that for now? We stick to `nodeSelector` alone and say these two types of traffic are indistinguishable. ?

Reply:
ok ! i have updated the NPEP now to outline and clarify this point a bit better,